r3bsource/trloii/R3BTrloiiSampReader.cxx

```diff
@@ -28,91 +28,100 @@ extern "C"
 #include "ext_h101_sampms.h"
 }
 
-using namespace std;
-
-R3BTrloiiSampReader::R3BTrloiiSampReader(EXT_STR_h101_SAMP* data, size_t offset)
+R3BTrloiiSampReader::R3BTrloiiSampReader(EXT_STR_h101_SAMP_onion* data, size_t offset)
     : R3BReader("R3BTrloiiSampReader")
     , fSampData(data)
+    , fSampLosData(nullptr)
+    , fSampMSData(nullptr)
+    , fSampLosMSData(nullptr)
     , fOffset(offset)
     , fSNum(1)
     , fSCNum(1)
     , fArray(new TClonesArray("R3BSamplerMappedData"))
+    , fArrayLH(nullptr)
     , fOnline(kFALSE)
 {
 }
 
-R3BTrloiiSampReader::R3BTrloiiSampReader(EXT_STR_h101_SAMPLOS* data, size_t offset)
+R3BTrloiiSampReader::R3BTrloiiSampReader(EXT_STR_h101_SAMPLOS_onion* data, size_t offset)
     : R3BReader("R3BTrloiiSampReader")
+    , fSampData(nullptr)
     , fSampLosData(data)
+    , fSampMSData(nullptr)
+    , fSampLosMSData(nullptr)
     , fOffset(offset)
     , fSNum(1)
     , fSCNum(2)
+    , fArray(nullptr)
     , fArrayLH(new TClonesArray("R3BSampLosMappedData"))
     , fOnline(kFALSE)
 {
 }
 
-R3BTrloiiSampReader::R3BTrloiiSampReader(EXT_STR_h101_SAMPMS* data, size_t offset)
+R3BTrloiiSampReader::R3BTrloiiSampReader(EXT_STR_h101_SAMPMS_onion* data, size_t offset)
     : R3BReader("R3BTrloiiSampMSReader")
+    , fSampData(nullptr)
+    , fSampLosData(nullptr)
     , fSampMSData(data)
+    , fSampLosMSData(nullptr)
     , fOffset(offset)
     , fSNum(2)
     , fSCNum(1)
     , fArray(new TClonesArray("R3BSamplerMappedData"))
+    , fArrayLH(nullptr)
     , fOnline(kFALSE)
 {
 }
 
-R3BTrloiiSampReader::R3BTrloiiSampReader(EXT_STR_h101_SAMPLOSMS* data, size_t offset)
+R3BTrloiiSampReader::R3BTrloiiSampReader(EXT_STR_h101_SAMPLOSMS_onion* data, size_t offset)
     : R3BReader("R3BTrloiiSampMSReader")
+    , fSampData(nullptr)
+    , fSampLosData(nullptr)
+    , fSampMSData(nullptr)
     , fSampLosMSData(data)
     , fOffset(offset)
     , fSNum(2)
     , fSCNum(2)
+    , fArray(nullptr)
     , fArrayLH(new TClonesArray("R3BSampLosMappedData"))
     , fOnline(kFALSE)
 {
 }
 
 R3BTrloiiSampReader::~R3BTrloiiSampReader()
 {
-    if (fArray)
-    {
-        delete fArray;
-    }
 
-    if (fArrayLH)
-    {
-        delete fArrayLH;
-    }
+    delete fArray;
+
+    delete fArrayLH;
```

Comment:
Output data is owned by FairRootManager. Deleting them sometimes could cause program crash.

Reply:
Agreed. 
First rule when dealing with ROOT: when in doubt, don't delete. 

At least if the lifetime of the object is about equal to the runtime of the program.